maker/src/main.rs

```diff
@@ -240,10 +240,14 @@ async fn main() -> Result<()> {
         p2p_socket,
     )?;
 
-    let (supervisor, price_feed) = supervisor::Actor::new(
-        bitmex_price_feed::Actor::new,
-        |_| true, // always restart price feed actor
-    );
+    let (supervisor, price_feed) =
+        supervisor::Actor::with_policy(bitmex_price_feed::Actor::default, |e| match e {
+            bitmex_price_feed::Error::FailedToParseQuote { .. }
+            | bitmex_price_feed::Error::Failed { .. }
+            | bitmex_price_feed::Error::FailedToConnect { .. }
+            | bitmex_price_feed::Error::Unspecified
+            | bitmex_price_feed::Error::StreamEnded => true, // always restart price feed actor
```

Comment:
❓ I guess we're explicitly matching because we want to reconsider our policy every time we modify these `Error`s?

Reply:
Yes but also the introduction of `UnitReason` required me to distinguish between the stop value `S` and the reason of the policy `R`. There is an `Into` requirement that only works if we know what the reason `R` is. By matching explicitly, we give the compiler a type-hint that `R` == `S` in this case and it doesn't whinge about unknown types.